mlflow/sklearn/__init__.py

```diff
@@ -948,6 +1041,52 @@ def patched_fit(original, self, *args, **kwargs):
                     FLAVOR_NAME, class_def, func_name, patched_fit, manage_run=True,
                 )
 
+        for func_name in ["predict", "transform", "predict_proba"]:
+            if hasattr(class_def, func_name):
+                safe_patch(
+                    FLAVOR_NAME, class_def, func_name, patched_predict, manage_run=False,
+                )
+
+    from sklearn import metrics
+    # Note:
+    #  don't need patch model.score() because internally model.score() will call the concrete metric API
+    for metric_method_name in metrics.__all__:
+        # excludes '**Display' object such as `ConfusionMatrixDisplay`
+        if callable(getattr(metrics, metric_method_name)) and not metric_method_name.endswith('Display'):
+            safe_patch(FLAVOR_NAME, metrics, metric_method_name, patched_metric_api, manage_run=False)
+
+    def patched_scorer_call(original, self, *args, **kwargs):
```

Comment:
Can we patch `_BaseScorer.__call__`?

https://github.com/scikit-learn/scikit-learn/blob/2beed55847ee70d363bdbfe14ee4401438fba057/sklearn/metrics/_scorer.py#L176

Reply:
seems `_BaseScorer` is not public interface.
And if user want define Scorer by themselves, it also doesn't require inherits from `_BaseScorer`